### PR TITLE
feat: smooth LiveView navigation instead of full page reloads

### PR DESCRIPTION
**Pull request by victor36max**
## Summary
- Replace all `window.location.assign()` calls with `liveSocket.js().navigate()` for SPA-like page transitions
- Add `navigate()` utility wrapper with fallback for safety (tests, edge cases)
- All 8 navigation sites across 5 React components now use smooth LiveView transitions that preserve the WebSocket connection

## Test plan
- [x] Navigate utility tests (4 tests covering liveSocket usage, options passthrough, fallback)
- [ ] Manual: click between projects, agents, settings, shared drive — verify no full page reload (no white flash, topbar progress shows)
- [ ] Manual: verify browser back/forward still works correctly

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: assets/react-components/lib/navigate.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Performs smooth LiveView navigation without a full page reload.
+ * Uses the Phoenix LiveView JS commands API (liveSocket.js().navigate).
+ * Falls back to window.location.assign if liveSocket is not available.
+ */
+export function navigate(href: string, opts?: { replace?: boolean }): void {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const liveSocket = (window as any).liveSocket;
```
Comment (victor36max) [inline assets/react-components/lib/navigate.ts]: if runtime check, don't use any

File: assets/test/navigate.test.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { describe, it, expect, vi, beforeEach, afterEach } from "vitest";
+import { navigate } from "../react-components/lib/navigate";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const win = window as unknown as Record<string, any>;
```
Comment (victor36max) [inline assets/test/navigate.test.ts]: avoid using eslint disable unless absolutely necessary, in this case can just be Record<string, unknown>
